crates/api_common/src/request.rs

```diff
@@ -217,62 +224,56 @@ pub async fn delete_image_from_pictrs(
   Ok(())
 }
 
-/// Both are options, since the URL might be either an html page, or an image
-/// Returns the SiteMetadata, and an image URL, if there is a picture associated
+/// Retrieves the image with local pict-rs and generates a thumbnail. Returns the thumbnail url.
 #[tracing::instrument(skip_all)]
-pub async fn fetch_site_data(
-  client: &ClientWithMiddleware,
-  settings: &Settings,
-  url: Option<&Url>,
-  include_image: bool,
-) -> (Option<SiteMetadata>, Option<DbUrl>) {
-  match &url {
-    Some(url) => {
-      // Fetch metadata
-      // Ignore errors, since it may be an image, or not have the data.
-      // Warning, this may ignore SSL errors
-      let metadata_option = fetch_site_metadata(client, url).await.ok();
-      if !include_image {
-        (metadata_option, None)
-      } else {
-        let thumbnail_url =
-          fetch_pictrs_url_from_site_metadata(client, &metadata_option, settings, url)
-            .await
-            .ok();
-        (metadata_option, thumbnail_url)
-      }
-    }
-    None => (None, None),
+async fn generate_pictrs_thumbnail(
+  image_url: &Url,
+  context: &LemmyContext,
+) -> Result<Url, LemmyError> {
+  let pictrs_config = context.settings().pictrs_config()?;
+
+  if !pictrs_config.cache_external_link_previews {
+    return Ok(
+      proxy_image_link(
```

Comment:
This block here is confusing now. Maybe we should remove `cache_external_link_previews` entirely now, and just leave `image_proxy`?